repos/system_upgrade/el7toel8/actors/opensshalgorithmscheck/actor.py

```diff
@@ -0,0 +1,60 @@
+from leapp.actors import Actor
+from leapp.models import Report, OpenSshConfig
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common.reporting import report_with_remediation
+
+
+class OpenSshAlgorithmsCheck(Actor):
+    """
+    OpenSSH configuration does not contain any unsupported cryptographic algorithms.
+
+    Check the values of Ciphers and MACs in OpenSSH server config file and warn
+    about removed algorithms which might cause the server to fail to start.
+    """
+    name = 'open_ssh_algorithms'
+    consumes = (OpenSshConfig,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        removed_ciphers = [
+            "blowfish-cbc",
+            "cast128-cbc",
+            "arcfour",
+            "arcfour128",
+            "arcfour256",
+        ]
+        removed_macs = [
+            "hmac-ripemd160",
+        ]
+        found_ciphers = []
+        found_macs = []
+        for config in self.consume(OpenSshConfig):
+            for cipher in removed_ciphers:
+                if config.ciphers and cipher in config.ciphers:
+                    found_ciphers.append(cipher)
+            for mac in removed_macs:
+                if config.macs and mac in config.macs:
+                    found_macs.append(mac)
+
+        if found_ciphers:
+            report_with_remediation(
+                title='OpenSSH configured to use removed ciphers',
+                summary='OpenSSH is configured to use removed ciphers %s. '
+                        'These ciphers were removed from OpenSSH and if '
+                        'present the sshd daemon will not start in RHEL 8'
+                        % ','.join(found_ciphers),
+                remediation='Remove the following ciphers from sshd_config: '
+                            '%s' % ','.join(found_ciphers),
+                severity='high')
+
+        if found_macs:
+            report_with_remediation(
+                title='OpenSSH configured to use removed mac',
+                summary='OpenSSH is configured to use removed mac %s. '
+                        'This MAC was removed from OpenSSH and if present '
+                        'the sshd daemon will not start in RHEL 8'
+                        % ','.join(found_macs),
+                remediation='Remove the following MACs from sshd_config: '
+                            '%s' % ','.join(found_macs),
+                severity='high')
```

Comment:
Shouldn't this report inhibit the upgrade?

Reply:
Fixed also this one